### ISSUE TITLE
Loom not instrumenting with latest Cadets LLVM

### ISSUE THREAD

**Issue by leestephenn** [cadets/loom#19]
I downloaded the most recent version of Loom and built the most recent version of LLVM from https://github.com/cadets/llvm/tree/cadets and https://github.com/cadets/clang/tree/cadets. While it builds fine, when I try to run it on a small file, I get the error:
    .../llvm-loom/Release/bin/opt: symbol lookup error: .../loom-raw/Release/lib/LLVMLoom.so: undefined symbol: _ZNK4llvm5Twine3strEv

In that and the rest of the commands, '...' stands for the path on my machine to the folder where I installed them.

The exact command I used was 
    .../llvm-loom/Release/bin/opt -load .../loom-raw/Release/lib/LLVMLoom.so -loom -loom-file really_small.policy -o Instreally_small.bc < really_small.opt

I have attached really_small.policy and really_small.opt, which was generated from the attached really_small.c. (All of them have .txt added to their file names to get GitHub to attach them.)

Note that my primary work uses an old version of Loom, so I can continue it while this remains open.

[really_small.c.txt](https://github.com/cadets/loom/files/2591147/really_small.c.txt)
[really_small.opt.txt](https://github.com/cadets/loom/files/2591145/really_small.opt.txt)
[really_small.policy.txt](https://github.com/cadets/loom/files/2591146/really_small.policy.txt)

**Comment by bkidney** [comment]
I saw something similar to this earlier this week that turned out to be a build cache issue. To solve it I deleted the build directory (Debug or Release) and ran cmake again. Could you try that?

**Comment by leestephenn** [comment]
I am now able to run the latest version of Loom again with the latest Cadets LLVM. The problem seems to have been that I compiled the Cadets LLVM with a different ABI for libstdc++ .

I have always needed to compile Loom with (some version of) Clang, because of [a known bug in G++ before version 7](https://stackoverflow.com/questions/25594644/warning-specialization-of-template-in-different-namespace). I, in contrast, compiled Cadets LLVM with G++ 5, which now uses [a different ABI](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html).

I fixed the problem by compiling both Cadets LLVM and Loom with LLVM 6.0.0 (the latest version I had installed). There probably is a way to compile Cadets LLVM with G++ and use the old ABI, but this sufficed and was simpler.
